ports/servoshell/minibrowser.rs

```diff
@@ -104,13 +109,15 @@ impl Minibrowser {
         }
     }
 
-    /// Updates the location field when the [Browser] says it has changed, unless the user has
-    /// started editing it without clicking Go.
+    /// Updates the location field from the given [Browser], unless the user has started editing it
+    /// without clicking Go, returning true iff the location has changed (needing an egui update).
     pub fn update_location_in_toolbar(&mut self, browser: &mut Browser<dyn WindowPortsMethods>) -> bool {
         if !self.location_dirty.get() {
             if let Some(location) = browser.current_url_string() {
-                self.location = RefCell::new(location.to_owned());
-                return true;
+                if location != self.location.get_mut() {
+                    self.location = RefCell::new(location.to_owned());
+                    return true;
+                }
```

Comment:
This becomes a bit deeply nested now. How do you feel about an early return in addition to a match statement?

```rust
// Comment explaining when this can happen, as it's still a little unclear
// to me and this is the opposite of what one might expect. A dirty variable
// usually triggers an update. Is it when the location is being edited?
if self.location_dirty.get() {
    return false;
}

match browser.current_url_string() {
    Some(location) if location != self.location.get_mut() {
        self.location = RefCell::new(location.to_owned());
        true
    }
   false
}
```

Reply:
done in b6fcf51e006f1801bf8b9aed8d31ab09a662c6b7